specifyweb/frontend/js_src/lib/components/AppResources/EditorComponents.tsx

```diff
@@ -201,15 +201,26 @@ export function useCodeMirrorExtensions(
       [appResource.specifyTable]
     )
   );
+
+  const [warnings, setWarnings] = React.useState<RA<Diagnostic>>([]);
+
   React.useEffect(() => {
-    const handleLinted = (results: RA<Diagnostic>): void =>
+    const handleLinted = (results: RA<Diagnostic>): void => {
+      setWarnings(
```

Comment:
not sure why, but the issue is caused by this setWarnings call (or rather, the re-render it causes)
possible, calling setWarnings triggers re-render at an inopportune time, and causes AppResourceTextEditor to hold a stale reference to code mirror, leading to no lint errors being visible (and if you look in the editor state, the are 0 errors there)

note that we are not using Codemirror directly, but using a react wrapper library - maybe it has a bug (though I can reproduce the issue on the latest version too)

I found a solution that works - calling openLintPanel directly in this file, rather than setting state - see the commit I pushed

Can you test if it behaves the same way in production (i.e react rendering twice may be causing this to happen)

Reply:
@maxxxxxdlp, it does not work. Neither on local. Why did you completely take off the warnings? I than cannot have that value in mu conditional in my useEffect to call displayWarningPanel

Reply:
the new push has the same behavior has how it is in xml-editor. But, we want the panel to open when opening the editor if warnings are present. Hence the call to openLintPanel in TabDefinition.

Reply:
works perfectly. Thank you!!!!